# Visualizacion 

Lo primero a realizar es la visualizacion de la tabla de datos de "P_BIOPRO.xpt".

In [1]:
import pandas as pd

df_p_biopro = pd.read_sas("P_BIOPRO.xpt")

df_p_biopro.head()

,SEQN,LBXSATSI,LBDSATLC,LBXSAL,LBDSALSI,LBXSAPSI,LBXSASSI,LBXSC3SI,LBXSBU,LBDSBUSI,...,LBXSCA,LBDSCASI,LBXSCH,LBDSCHSI,LBXSTP,LBDSTPSI,LBXSTR,LBDSTRSI,LBXSUA,LBDSUASI
0,109264.0,8.0,5.397605e-79,4.3,43.0,172.0,18.0,26.0,9.0,3.21,...,9.4,2.350,170.0,4.396,6.9,69.0,54.0,0.610,3.5,208.2
1,109266.0,15.0,5.397605e-79,3.8,38.0,74.0,14.0,25.0,8.0,2.86,...,8.8,2.200,199.0,5.146,6.8,68.0,86.0,0.971,4.9,291.5
2,109271.0,8.0,5.397605e-79,3.8,38.0,88.0,14.0,30.0,8.0,2.86,...,8.9,2.225,148.0,3.827,7.3,73.0,101.0,1.140,5.0,297.4
3,109273.0,35.0,5.397605e-79,4.6,46.0,70.0,27.0,27.0,11.0,3.93,...,9.7,2.425,168.0,4.344,7.4,74.0,178.0,2.010,5.2,309.3
4,109274.0,19.0,5.397605e-79,4.0,40.0,49.0,20.0,28.0,11.0,3.93,...,9.1,2.275,105.0,2.715,6.8,68.0,151.0,1.705,5.1,303.3


y observamos cada uno de sus atributos

In [2]:
df_p_biopro.columns

Index(['SEQN', 'LBXSATSI', 'LBDSATLC', 'LBXSAL', 'LBDSALSI', 'LBXSAPSI',
       'LBXSASSI', 'LBXSC3SI', 'LBXSBU', 'LBDSBUSI', 'LBXSCLSI', 'LBXSCK',
       'LBXSCR', 'LBDSCRSI', 'LBXSGB', 'LBDSGBSI', 'LBXSGL', 'LBDSGLSI',
       'LBXSGTSI', 'LBDSGTLC', 'LBXSIR', 'LBDSIRSI', 'LBXSLDSI', 'LBXSOSSI',
       'LBXSPH', 'LBDSPHSI', 'LBXSKSI', 'LBXSNASI', 'LBXSTB', 'LBDSTBSI',
       'LBDSTBLC', 'LBXSCA', 'LBDSCASI', 'LBXSCH', 'LBDSCHSI', 'LBXSTP',
       'LBDSTPSI', 'LBXSTR', 'LBDSTRSI', 'LBXSUA', 'LBDSUASI'],
      dtype='str')

A pesar de tener una nomenclatura bastante revuelta, esta se expresa de la siguiente manera:

<ol>
    <li><b>SEQN</b>: Respondent Sequence Number</li>
    <li><b>LBXSATSI</b>: Alanine Aminotransferase (ALT) (U/L)</li>
    <li><b>LBDSATLC</b>: Alanine Aminotransferase (ALT) Comment Code</li>
    <li><b>LBXSAL</b>: Albumin, refrigerated serum (g/dL)</li>
    <li><b>LBDSALSI</b>: Albumin, refrigerated serum(g/L)</li>
    <li><b>LBXSAPSI</b>: Alkaline Phosphatase (ALP) (IU/L)</li>
    <li><b>LBXSASSI</b>: Aspartate Aminotransferase (AST) (U/L)</li>
    <li><b>LBXSC3SI</b>: Bicarbonate (mmol/L)</li>
    <li><b>LBXSBU</b>: Blood Urea Nitrogen (mg/dL)</li>
    <li><b>LBDSBUSI</b>: Blood Urea Nitrogen (mmol/L)</li>
    <li><b>LBXSCLSI</b>: Chloride (mmol/L)</li>
    <li><b>LBXSCK</b>: Creatine Phosphokinase (CPK) (IU/L)</li>
    <li><b>LBXSCR</b>: Creatinine, refrigerated serum (mg/dL)</li>
    <li><b>LBDSCRSI</b>: Creatinine, refrigerated serum (umol/L)</li>
    <li><b>LBXSGB</b>: Globulin (g/dL)</li>
    <li><b>LBDSGBSI </b>: Globulin (g/L)</li>
    <li><b>LBXSGL</b>: Glucose, refrigerated serum (mg/dL)</li>
    <li><b>LBDSGLSI</b>: Glucose, refrigerated serum (mmol/L)</li>
    <li><b>LBXSGTSI</b>: Gamma Glutamyl Transferase (GGT) (IU/L)</li>
    <li><b>LBDSGTLC</b>: Gamma Glutamyl Transferase (GGT) Comment Code</li>
    <li><b>LBXSIR</b>: Iron, refrigerated serum (ug/dL)</li>
    <li><b>LBDSIRSI</b>: Iron, refrigerated serum (umol/L)</li>
    <li><b>LBXSLDSI</b>: Lactate Dehydrogenase (LDH) (IU/L)</li>
    <li><b>LBXSOSSI</b>: Osmolality (mmol/Kg)</li>
    <li><b>LBXSPH</b>: Phosphorus (mg/dL)</li>
    <li><b>LBDSPHSI</b>: Phosphorus (mmol/L)</li>
    <li><b>LBXSKSI</b>: Potassium (mmol/L)</li>
    <li><b>LBXSNASI</b>: Sodium (mmol/L)</li>
    <li><b>LBXSTB</b>: Total Bilirubin (mg/dL)</li>
    <li><b>LBDSTBSI</b>: Total Bilirubin (umol/L)</li>
    <li><b>LBDSTBLC</b>: Total Bilirubin Comment Code</li>
    <li><b>LBXSCA</b>: Total Calcium (mg/dL)</li>
    <li><b>LBDSCASI</b>: Total Calcium (mmol/L)</li>
    <li><b>LBXSCH</b>: Cholesterol, refrigerated serum (mg/dL)</li>
    <li><b>LBDSCHSI</b>: Cholesterol, refrigerated serum (mmol/L)</li>
    <li><b>LBXSTP</b>: Total Protein (g/dL)</li>
    <li><b>LBDSTPSI</b>: Total Protein (g/L)</li>
    <li><b>LBXSTR</b>: Triglycerides, refrig serum (mg/dL)</li>
    <li><b>LBDSTRSI</b>: Triglycerides, refrig serum (mmol/L)</li>
    <li><b>LBXSUA</b>: Uric acid (mg/dL)</li>
    <li><b>LBDSUASI</b>: Uric acid (umol/L)</li>
</ol>

In [3]:
df_p_biopro.shape

(10409, 41)

# Depuracion 

En la tabla "P_BIOPRO.xpt" se repiten columnas con los aributos, pero sus unidades de medida son las que cambian, siendo este cambio en factores de 100 o 1000, lo que quiere decir que este puede estar en mmg/dL o mmol/L, lo cual solo afecta el valor numerico pero no el analisis en sí mismo. Las causas por las que se modificara la tabla son:

<ol>
    <li>Se eliminan las columnas que tengan como unico dato al valor nulo</li>
    <li>Se eliminan las columnas que posean mismos atributos pero con distinta unidad de medida, conservando aquellas que posean el SI</li>
    <li>Se eliminan las filas que como unico dato tengan el SEQN</li>
</ol>

Con esta idea se logra una mejor depuracion de los datos

In [4]:
import pandas as pd

# 1. Cargar el archivo .xpt
nombre_entrada = 'P_BIOPRO.xpt'
df = pd.read_sas(nombre_entrada)

# --- PASO A: ELIMINAR COLUMNAS ESPECÍFICAS ---
columnas_a_eliminar = [
    'LBXSAL', 'LBXSBU', 'LBXSCR', 'LBXSGB', 'LBXSGL', 'LBXSIR', 'LBXSPH', 'LBXSTB', 'LBXSCA', 'LBXSCH',
    'LBXSTP', 'LBXSTR', 'LBXSUA', 'LBDSATLC', 'LBDSTBLC', 'LBDSGTLC'
]
df_limpio = df.drop(columns=columnas_a_eliminar, errors='ignore')

# --- PASO A.1: ELIMINAR COLUMNAS SI TODA LA COLUMNA ES NULA ---
# axis=1 indica columnas, how='all' indica que TODOS los valores deben ser NaN
df_limpio = df_limpio.dropna(axis=1, how='all')

# --- PASO B: ELIMINAR FILAS VACÍAS ---
# Definimos qué columnas usaremos para evaluar si la fila está "vacía"
# (Usamos las columnas actuales de df_limpio)
columnas_datos = [col for col in df_limpio.columns if col != 'SEQN']

# how="all" significa que solo borra la fila si TODAS las columnas de datos son NaN
df_limpio = df_limpio.dropna(subset=columnas_datos, how='all')

# --- PASO C: GUARDAR RESULTADOS ---
nombre_salida_csv = 'P_BIOPRO_limpio.csv'
df_limpio.to_csv(nombre_salida_csv, index=False, encoding='utf-8-sig')

# --- REPORTE DE CAMBIOS ---
print("-" * 30)
print("REPORTE DE LIMPIEZA FINAL")
print("-" * 30)
print(f"Filas originales: {len(df)}")
print(f"Filas resultantes: {len(df_limpio)}")
print(f"Columnas originales: {len(df.columns)}")
print(f"Columnas eliminadas por lista: {len(columnas_a_eliminar)}")
print(f"Columnas finales: {len(df_limpio.columns)}")
print(f"Archivo guardado como: {nombre_salida_csv}")

------------------------------
REPORTE DE LIMPIEZA FINAL
------------------------------
Filas originales: 10409
Filas resultantes: 9477
Columnas originales: 41
Columnas eliminadas por lista: 16
Columnas finales: 25
Archivo guardado como: P_BIOPRO_limpio.csv


Inicialmente eran 10409 filas de datos y 41 columnas, pero luego de la depuracion se tienen 9477 filas con 25 columnas, con datos efectivos para poder trabajarlos sin ningun problema. Al final arroja una nueva tabla llamada "P_BIOPRO_limpio.csv". Algo importante a observar es que la extension del archivo cambio a uno mas amigable

In [5]:
import pandas as pd

df_p_biopro_limpio = pd.read_csv("P_BIOPRO_limpio.csv")

df_p_biopro_limpio.head()

,SEQN,LBXSATSI,LBDSALSI,LBXSAPSI,LBXSASSI,LBXSC3SI,LBDSBUSI,LBXSCLSI,LBXSCK,LBDSCRSI,...,LBXSOSSI,LBDSPHSI,LBXSKSI,LBXSNASI,LBDSTBSI,LBDSCASI,LBDSCHSI,LBDSTPSI,LBDSTRSI,LBDSUASI
0,109264.0,8.0,43.0,172.0,18.0,26.0,3.21,102.0,48.0,50.39,...,276.0,1.453,4.0,139.0,8.55,2.350,4.396,69.0,0.610,208.2
1,109266.0,15.0,38.0,74.0,14.0,25.0,2.86,99.0,109.0,55.69,...,277.0,1.033,3.6,140.0,8.55,2.200,5.146,68.0,0.971,291.5
2,109271.0,8.0,38.0,88.0,14.0,30.0,2.86,101.0,85.0,68.95,...,279.0,1.292,4.3,141.0,8.55,2.225,3.827,73.0,1.140,297.4
3,109273.0,35.0,46.0,70.0,27.0,27.0,3.93,100.0,48.0,81.33,...,276.0,0.872,4.3,139.0,10.26,2.425,4.344,74.0,2.010,309.3
4,109274.0,19.0,40.0,49.0,20.0,28.0,3.93,99.0,93.0,65.42,...,278.0,0.872,3.9,138.0,10.26,2.275,2.715,68.0,1.705,303.3


Tambien, al comparar las columnas del archivo original al nuevo se observa una disminucion de atributos considerable, siendo ahora 25 atributos para poder trabajar

In [6]:
df_p_biopro.columns

Index(['SEQN', 'LBXSATSI', 'LBDSATLC', 'LBXSAL', 'LBDSALSI', 'LBXSAPSI',
       'LBXSASSI', 'LBXSC3SI', 'LBXSBU', 'LBDSBUSI', 'LBXSCLSI', 'LBXSCK',
       'LBXSCR', 'LBDSCRSI', 'LBXSGB', 'LBDSGBSI', 'LBXSGL', 'LBDSGLSI',
       'LBXSGTSI', 'LBDSGTLC', 'LBXSIR', 'LBDSIRSI', 'LBXSLDSI', 'LBXSOSSI',
       'LBXSPH', 'LBDSPHSI', 'LBXSKSI', 'LBXSNASI', 'LBXSTB', 'LBDSTBSI',
       'LBDSTBLC', 'LBXSCA', 'LBDSCASI', 'LBXSCH', 'LBDSCHSI', 'LBXSTP',
       'LBDSTPSI', 'LBXSTR', 'LBDSTRSI', 'LBXSUA', 'LBDSUASI'],
      dtype='str')

In [7]:
df_p_biopro_limpio.columns

Index(['SEQN', 'LBXSATSI', 'LBDSALSI', 'LBXSAPSI', 'LBXSASSI', 'LBXSC3SI',
       'LBDSBUSI', 'LBXSCLSI', 'LBXSCK', 'LBDSCRSI', 'LBDSGBSI', 'LBDSGLSI',
       'LBXSGTSI', 'LBDSIRSI', 'LBXSLDSI', 'LBXSOSSI', 'LBDSPHSI', 'LBXSKSI',
       'LBXSNASI', 'LBDSTBSI', 'LBDSCASI', 'LBDSCHSI', 'LBDSTPSI', 'LBDSTRSI',
       'LBDSUASI'],
      dtype='str')

In [8]:
df_p_biopro.shape

(10409, 41)

In [9]:
df_p_biopro_limpio.shape

(9477, 25)

# Cambio a nombres mas legibles 

Ahora, se cambian los nombres de los atributos a unos mas legibles

In [10]:
# 1. Diccionario de mapeo específico para las columnas que conservaste (unidades SI)
mapping_nombres = {
    'SEQN': 'ID_Participante',
    'LBXSATSI': 'ALT_U_L (Alanina Aminotransferasa)',
    'LBDSALSI': 'Albumina_g_L',
    'LBXSAPSI': 'Fosfatasa_Alcalina_U_L',
    'LBXSASSI': 'AST_U_L (Aspartato Aminotransferasa)',
    'LBXSC3SI': 'Bicarbonato_mmol_L',
    'LBDSBUSI': 'Nitrogeno_Ureico_mmol_L (BUN)',
    'LBXSCLSI': 'Cloruro_mmol_L',
    'LBXSCK': 'Creatina_Quinasa_U_L',
    'LBDSCRSI': 'Creatinina_umol_L',
    'LBDSGBSI': 'Globulina_g_L',
    'LBDSGLSI': 'Glucosa_mmol_L',
    'LBXSGTSI': 'GGT_U_L (Gamma-glutamil Transaminasa)',
    'LBDSIRSI': 'Hierro_umol_L',
    'LBXSLDSI': 'LDH_U_L (Lactato Deshidrogenasa)',
    'LBXSOSSI': 'Osmolalidad_mmol_Kg',
    'LBDSPHSI': 'Fosforo_mmol_L',
    'LBXSKSI': 'Potasio_mmol_L',
    'LBXSNASI': 'Sodio_mmol_L',
    'LBDSTBSI': 'Bilirrubina_Total_umol_L',
    'LBDSCASI': 'Calcio_Total_mmol_L',
    'LBDSCHSI': 'Colesterol_mmol_L',
    'LBDSTPSI': 'Proteinas_Totales_g_L',
    'LBDSTRSI': 'Trigliceridos_mmol_L',
    'LBDSUASI': 'Acido_Urico_umol_L'
}

# 2. Renombrar las columnas en el DataFrame limpio
df_p_biopro_final = df_limpio.rename(columns=mapping_nombres)

# 3. Exportar a un nuevo archivo CSV
nombre_archivo_humano = 'P_BIOPRO_nombres_claros.csv'
df_p_biopro_final.to_csv(nombre_archivo_humano, index=False, encoding='utf-8-sig')

print(f"✅ Proceso completado. El archivo '{nombre_archivo_humano}' ha sido generado con nombres legibles.")
print(f"Total de columnas procesadas: {len(df_p_biopro_final.columns)}")

✅ Proceso completado. El archivo 'P_BIOPRO_nombres_claros.csv' ha sido generado con nombres legibles.
Total de columnas procesadas: 25


Nuevamente se visualiza la nueva tabla generada

In [11]:
import pandas as pd

df_p_biopro_nombres_claros = pd.read_csv("P_BIOPRO_nombres_claros.csv")

df_p_biopro_nombres_claros.head()

,ID_Participante,ALT_U_L (Alanina Aminotransferasa),Albumina_g_L,Fosfatasa_Alcalina_U_L,AST_U_L (Aspartato Aminotransferasa),Bicarbonato_mmol_L,Nitrogeno_Ureico_mmol_L (BUN),Cloruro_mmol_L,Creatina_Quinasa_U_L,Creatinina_umol_L,...,Osmolalidad_mmol_Kg,Fosforo_mmol_L,Potasio_mmol_L,Sodio_mmol_L,Bilirrubina_Total_umol_L,Calcio_Total_mmol_L,Colesterol_mmol_L,Proteinas_Totales_g_L,Trigliceridos_mmol_L,Acido_Urico_umol_L
0,109264.0,8.0,43.0,172.0,18.0,26.0,3.21,102.0,48.0,50.39,...,276.0,1.453,4.0,139.0,8.55,2.350,4.396,69.0,0.610,208.2
1,109266.0,15.0,38.0,74.0,14.0,25.0,2.86,99.0,109.0,55.69,...,277.0,1.033,3.6,140.0,8.55,2.200,5.146,68.0,0.971,291.5
2,109271.0,8.0,38.0,88.0,14.0,30.0,2.86,101.0,85.0,68.95,...,279.0,1.292,4.3,141.0,8.55,2.225,3.827,73.0,1.140,297.4
3,109273.0,35.0,46.0,70.0,27.0,27.0,3.93,100.0,48.0,81.33,...,276.0,0.872,4.3,139.0,10.26,2.425,4.344,74.0,2.010,309.3
4,109274.0,19.0,40.0,49.0,20.0,28.0,3.93,99.0,93.0,65.42,...,278.0,0.872,3.9,138.0,10.26,2.275,2.715,68.0,1.705,303.3
